sdk/web-pubsub/web-pubsub-express/src/cloudEventsDispatcher.ts

```diff
@@ -232,13 +258,13 @@ export class CloudEventsDispatcher {
         return false;
     }
 
-    const receivedEvent = await utils.convertHttpToEvent(request);
-    logger.verbose(receivedEvent);
-
     switch (eventType) {
       case EventType.Connect: {
-        const connectRequest = receivedEvent.data as ConnectRequest;
-        connectRequest.context = getContext(receivedEvent, origin);
+        const body = (await utils.readRequestBody(request)).toString();
+        const connectRequest = JSON.parse(body) as ConnectRequest;
```

Comment:
As a small convenience, perhaps a function like

```ts
async function readRequest<T>(request: IncomingMessage): Promise<T> {
   const body = (await utils.readRequestBody(request)).toString();
   return JSON.parse(body) as T;
}
```

so you can write it like

```ts
const connectRequest = await readRequest<ConnectRequest>(request);
connectRequest.context = getContext(request, origin);
```

Or if we wanted to get even fancier so it can handle both types of events:

```ts
async function readRequest(request:IncomingMessage, dataType: string): Promise<UserEventRequest>;
async function readRequest<T extends {context: ConnectionContext}>(request:IncomingMessage): Promise<T>;
async function readRequest<T extends {context: ConnectionContext}>(request: IncomingMessage, dataType?: string): Promise<T | UserEventRequest> {
   const context = getContext(request, origin);
   const body = (await utils.readRequestBody(request)).toString();
   const data =  JSON.parse(body) as T;
   if (dataType) {
       return { context, data, dataType };
   } else {
       data.context = context;
       return data;
   }
}
```